api/src/main/java/me/lokka30/treasury/api/economy/account/Account.java

```diff
@@ -152,4 +153,13 @@ public void fail(@NotNull EconomyException exception) {
      */
     void deleteAccount(@NotNull EconomySubscriber<Boolean> subscription);
 
+    /**
+     * Returns the {@link Currency Currencies} this {@code Account} has balance in.
+     *
+     * @param subscription the {@link EconomySubscriber} accepting the currencies
+     * @author MrIvanPlays
+     * @since {@link me.lokka30.treasury.api.economy.misc.EconomyAPIVersion#v1_0 v1.0}
+     */
+    void retrieveCurrenciesById(@NotNull EconomySubscriber<Collection<UUID>> subscription);
```

Comment:
It is possible that the economy plugin stores every currency for an account even if they have $0 in it. This is what my economy plugin will do. Then, what should constitute a provider listing a currency in this method? Otherwise all currencies can be just returned.

Reply:
Perhaps `retrieveHeldCurrencies` is a better name for the method. I am unsure about it

Reply:
> It is possible that the economy plugin stores every currency for an account even if they have $0 in it.

Sure but why should you? 

> what should constitute a provider listing a currency in this method?

A provider should list all the currencies this account has if the currency's balance is > 0

Reply:
Fair points. We need to note that economies with negative balance support should consider all currencies being held by every account then.